source/common/request_info/dynamic_metadata_impl.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <map>
+
+#include "envoy/request_info/dynamic_metadata.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+class DynamicMetadataImpl : public DynamicMetadata {
+public:
+  DynamicMetadataImpl() {}
+  ~DynamicMetadataImpl() override;
+
+  // DynamicMetadata
+  void setDataGeneric(absl::string_view data_name, size_t type_id, void* data,
+                      void (*destructor)(void*)) override;
+  void* getDataGeneric(absl::string_view data_name, size_t type_id) const override;
+  bool hasDataGeneric(absl::string_view data_name, size_t type_id) const override;
+  bool hasDataWithName(absl::string_view data_name) const override;
+
+private:
+  struct Data {
+    size_t typeid_;
+    void* ptr_;
+    void (*destructor_)(void*);
+  };
+
+  std::map<std::string, Data, std::less<>> data_storage_;
```

Comment:
Nit: is there a reason this is ordered? Since we're not traversing, I don't know if there is a strong case here.

Reply:
This turns out to be necessary in order to use an absl::string_view as a find() argument.  See https://en.cppreference.com/w/cpp/container/map/find, specifically instances 3 & 4 of the method.